mmv1/third_party/terraform/services/resourcemanager/resource_google_project_iam_member_remove.go

```diff
@@ -0,0 +1,109 @@
+package resourcemanager
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgiamresource"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+	"google.golang.org/api/cloudresourcemanager/v1"
+)
+
+func ResourceGoogleProjectIamMemberRemove() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGoogleProjectIamMemberRemoveCreate,
+		Read:   resourceGoogleProjectIamMemberRemoveRead,
+		Delete: resourceGoogleProjectIamMemberRemoveDelete,
+
+		CustomizeDiff: customdiff.All(
+			tpgresource.DefaultProviderProject,
+		),
+
+		Schema: map[string]*schema.Schema{
+			"role": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"project": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The project that the service account will be created in. Defaults to the provider project configuration.`,
+			},
+			"member": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: `The Identity of the service account in the form 'serviceAccount:{email}'. This value is often used to refer to the service account in order to grant IAM permissions.`,
+			},
+		},
+		UseJSONNumber: true,
+	}
+}
+
+func resourceGoogleProjectIamMemberRemoveCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*transport_tpg.Config)
+
+	project := d.Get("project").(string)
+	role := d.Get("role").(string)
+	member := d.Get("member").(string)
+
+	iamPolicy, err := config.NewResourceManagerClient(config.UserAgent).Projects.GetIamPolicy(project,
+		&cloudresourcemanager.GetIamPolicyRequest{
+			Options: &cloudresourcemanager.GetPolicyOptions{
+				RequestedPolicyVersion: tpgiamresource.IamPolicyVersion,
+			},
+		}).Do()
+	for _, bind := range iamPolicy.Bindings {
+		for _, existingMember := range bind.Members {
+			if member == existingMember {
+				if role == bind.Role {
+					bind.Role = "role/remove"
```

Comment:
This creates the right request, but I'm not entirely sure why it does- can you point to docs or code that makes this work? I expected an API error on first read.

Reply:
According to the logs, the role is removed. https://cloud.google.com/resource-manager/reference/rest/v1/projects/setIamPolicy. On the other hand it might not be the best message to send.

Reply:
I don't see why we're adding role/remove on that page, though, rather than removing the role from the binding (or the binding altogether)

How would this work if there were more members in the same binding, as well?

Reply:
remove role involves a GetIamPolicy request and then a SetIamPolicy request. https://yaqs.corp.google.com/eng/q/7281177105365204992. 

For this specific case, we only consider member as a string, not a set. So only one binding could be removed at a time.

Reply:
> So only one binding could be removed at a time.

That's correct- we can only remove one binding at a time, and only if we just removed the last member from the binding.

We should verify we are not removing the entire binding when we are removing only a single member in a binding with multiple members.

> remove role involves a GetIamPolicy request and then a SetIamPolicy request

Not sure I understand. We are doing a GetIamPolicy and then a SetIamPolicy request.

Reply:
removed this line as it caused some API errors.